UploadServiceLogo/handler.ts

```diff
@@ -0,0 +1,136 @@
+import { isNone } from "fp-ts/lib/Option";
+
+import {
+  IResponseErrorInternal,
+  IResponseErrorNotFound,
+  IResponseErrorValidation,
+  IResponseSuccessRedirectToResource,
+  ResponseErrorNotFound,
+  ResponseSuccessRedirectToResource
+} from "italia-ts-commons/lib/responses";
+
+import { ServiceModel } from "io-functions-commons/dist/src/models/service";
+import { CustomTelemetryClientFactory } from "io-functions-commons/dist/src/utils/application_insights";
+import {
+  AzureApiAuthMiddleware,
+  IAzureApiAuthorization,
+  UserGroup
+} from "io-functions-commons/dist/src/utils/middlewares/azure_api_auth";
+import {
+  AzureUserAttributesMiddleware,
+  IAzureUserAttributes
+} from "io-functions-commons/dist/src/utils/middlewares/azure_user_attributes";
+import {
+  ClientIp,
+  ClientIpMiddleware
+} from "io-functions-commons/dist/src/utils/middlewares/client_ip_middleware";
+import { ContextMiddleware } from "io-functions-commons/dist/src/utils/middlewares/context_middleware";
+import {
+  withRequestMiddlewares,
+  wrapRequestHandler
+} from "io-functions-commons/dist/src/utils/request_middleware";
+import {
+  IResponseErrorQuery,
+  ResponseErrorQuery
+} from "io-functions-commons/dist/src/utils/response";
+import {
+  checkSourceIpForHandler,
+  clientIPAndCidrTuple as ipTuple
+} from "io-functions-commons/dist/src/utils/source_ip_check";
+
+import { Logo as ApiLogo } from "../generated/definitions/Logo";
+import { ServiceId } from "../generated/definitions/ServiceId";
+import { LogoPayloadMiddleware } from "../utils/middlewares/service";
+import { ServiceIdMiddleware } from "../utils/middlewares/serviceid";
+
+type IUpdateServiceHandler = (
+  context: Context,
+  auth: IAzureApiAuthorization,
+  clientIp: ClientIp,
+  userAttributes: IAzureUserAttributes,
+  serviceId: ServiceId,
+  logoPayload: ApiLogo
+) => Promise<
+  // tslint:disable-next-line: max-union-size
+  | IResponseSuccessRedirectToResource<{}, {}>
+  | IResponseErrorValidation
+  | IResponseErrorQuery
+  | IResponseErrorNotFound
+  | IResponseErrorInternal
+>;
+
+export function UpdateServiceLogoHandler(
+  _GCTC: CustomTelemetryClientFactory,
+  serviceModel: ServiceModel,
+  logosHost: string
+): IUpdateServiceHandler {
+  return async (context, __, ___, ____, serviceId, logoPayload) => {
+    const errorOrMaybeRetrievedService = await serviceModel.findOneByServiceId(
+      serviceId
+    );
+    if (isLeft(errorOrMaybeRetrievedService)) {
+      return ResponseErrorQuery(
+        "Error trying to retrieve existing service",
+        errorOrMaybeRetrievedService.value
+      );
+    }
+
+    const maybeService = errorOrMaybeRetrievedService.value;
+    if (isNone(maybeService)) {
+      return ResponseErrorNotFound(
+        "Error",
+        "Could not find a service with the provided serviceId"
+      );
+    }
+
+    // tslint:disable-next-line:no-object-mutation
+    context.bindings.logo = Buffer.from(logoPayload.logo, "base64");
```

Comment:
can `Buffer.from()` fail ?

Reply:
it looks like you have to wrap it around a tryCatch() block:
https://nodejs.org/api/buffer.html#buffer_class_method_buffer_from_array

Reply:
It shouldn't be necessary, because according to the documentation a TypeError will be thrown if `logoPayload.logo` is not an appropriate type, but its type is already checked and validated by the [LogoPayload middleware ](https://github.com/pagopa/io-functions-admin/pull/10/files/ee3cc68f0ff5181b129f8b466665de846f3bd95e#diff-3a06dc6a4c785efd4b3fff428ad19a58R23) inside the [middlewaresWrap](https://github.com/pagopa/io-functions-admin/pull/10/files/ee3cc68f0ff5181b129f8b466665de846f3bd95e#diff-c1bc7bbe2aef8853a2b62bbb31e69937R116).